### PR TITLE
Add documentation for prov.py

### LINKED ISSUE #187: Add documentation to StepRetroProv & WorkflowRetroProv
The classes in `prov.py` still need a lot of documentation.

### PR DESCRIPTION
**Pull request by svenvanderburg**
Fixes #187 and I couldn't resist doing some refactoring as well.

### CODE REVIEW

**Review by raar1 — APPROVED** [review]
Nice work. Just one query about set_attribute below.

File: fairworkflows/prov.py [hunk 1]
```diff
@@ -110,10 +138,31 @@ def __str__(self):
...
+            workflow_uri: uri of the prospective workflow (NB: we do not use workflow.uri for
+                this because the workflow can be unpublished.)
+            step_provs: A list of StepRetroProv objects for each individual step of the workflow.
+
+        """
         super().__init__(uri=None, ref_name='fairworkflowprov')
-        self._rdf.add((self.self_ref, rdflib.RDF.type, namespaces.PPLAN.Bundle))
-        self._rdf.add((self.self_ref, rdflib.RDF.type, namespaces.PROV.Collection))
+        self.set_attribute(rdflib.RDF.type, namespaces.PPLAN.Bundle)
+        self.set_attribute(rdflib.RDF.type, namespaces.PROV.Collection)
```
[the remark this answers is not in the document]
Comment (svenvanderburg) [reply fairworkflows/prov.py]: Yes it does! Very good point. I solved it by setting `overwrite=False` ;)

[the remark this answers is not in the document]
Comment (svenvanderburg) [reply fairworkflows/prov.py]: Also at some other places.

[the remark this answers is not in the document]
Comment (svenvanderburg) [reply fairworkflows/prov.py]: I checked the result in the notebook, it looks good.

**Pull request merged**